loops/driver.js

```diff
@@ -123,7 +123,9 @@ function pauseGame(ping) {
 function prepareRestart() {
     const curAction = actions.getNextValidAction();
     if (options.pauseBeforeRestart ||
-        (options.pauseOnFailedLoop && actions.current.filter(action => action.loopsLeft > 0).length > 0)) {
+        (options.pauseOnFailedLoop && 
+         (actions.current.filter(action => action.loopsFailed > 0).length > 0 ||
```

Comment:
Is the `loopsFailed` really needed here? If you fail an action, the action will always have at least 1 loop left since you either couldn't finish the last loop, or were never able to start the action in the first place. I could be missing something though.

Reply:
actions.getNextValidAction does this when an action fails due to wrong zone or missing resources:
```
            curAction.loopsFailed = curAction.loopsLeft;
            curAction.loopsLeft = 0;
```
So checking loopsLeft wouldn't catch it.  I guess we could delete that and skip the loopsFailed check...